glean_parser/util.py

```diff
@@ -346,29 +365,24 @@ def is_expired(expires: str) -> bool:
     elif expires == "expired":
         return True
     else:
-        try:
-            if sys.version_info < (3, 7):
-                date = iso8601.parse_date(expires).date()
-            else:
-                date = datetime.date.fromisoformat(expires)
-        except ValueError:
-            raise ValueError(
-                f"Invalid expiration date '{expires}'. "
-                "Must be of the form yyyy-mm-dd in UTC."
-            )
+        date = parse_expires(expires)
         return date <= datetime.datetime.utcnow().date()
 
 
 def validate_expires(expires: str) -> None:
     """
-    Raises ValueError if `expires` is not valid.
+    Raises a ValueError in case the date is more than 730 days (~2 years) in the future
     """
     if expires in ("never", "expired"):
         return
-    if sys.version_info < (3, 7):
-        iso8601.parse_date(expires)
-    else:
-        datetime.date.fromisoformat(expires)
+
+    date = parse_expires(expires)
+    max_date = datetime.datetime.now() + datetime.timedelta(days=730)
+    if date > max_date.date():
+        raise ValueError(
+            f"Invalid expiration date '{expires}'. '"
+            "Date must not be more than 730 days (~2 years) in the future."
```

Comment:
I think we should be a little kinder in the language here. A date more than 2 years in the future isn't invalid as much as it is a likely error.

Perhaps `"Expiration date '{expires}' is more than 730 days (2 years) in the future. Please make sure this is intentional."`? What do you think?

Reply:
FWIW, there is a message and the bottom of `glinter` output about the `no_lint` option that's always displayed when there are 1 or more warnings or errors.  So *maybe* no need to explicitly have that here, unless we **really** want to highlight that.

Reply:
Uh oh, this was alredy merged with the message and I was not aware of that. I can quickly remove the message if you think its best. https://github.com/mozilla/glean_parser/blob/main/glean_parser/util.py#L386-L387

Reply:
Not critical -- I didn't read the final diff when I wrote the comment.  We can remove later if it annoys anyone (but it probably won't).